### ISSUE TITLE
About training code

### ISSUE THREAD

**Issue by yuhu-dsp** [v1t0ry/ICAP#1]
I tried to train the prior hyper-parameters with mini-batch gradient ascent algorithm, but it turns out that my parameters are kind of different from yours, or the original cap. And the results are of course worse than yours. Can't find any problems in my code. Wondering if you could give some advise about the training process, or just share part of your code for training. That would be a lot to me. Thx.

**Comment by datngo93** [comment]
There are several factors that affect the training process.

First of all, I would like to inform you that it is impossible to obtain the same parameters as mine or Zhu et al.'s (in the original CAP) because the depth map values are drawn from a random number generator. However, among three parameters, the bias and brightness coefficients are typically positive, whereas the saturation coefficient is typically negative. This could give a first glance whether the training is as intended.

Secondly, it is also advisable to check the log-likelihood value to verify whether it has reached the plateau. Also, by investigating the graph, hyper-parameters can be tuned accordingly. As an example, I attached a log-likelihood graph when I applied maximum likelihood estimates to a training dataset containing only 20 images.
![mle_20](https://user-images.githubusercontent.com/8115608/111103514-fb6e6400-8591-11eb-9558-c240405a0fc7.png)

Finally, I would like to share my training code with you and I hope that it would be useful to you. In the code, I implemented the same mini-batch gradient ascent algorithm but the number of parameters is 4 instead of 3 (because I added the image's hue information to the linear model, so that d = theta0 + theta1 x value + theta2 x saturation + theta3 x hue). Since I do not have enough time to trim the code, you are suggested to remove theta3 and revise the code before performing training. Also, the input data to my code is 1D column vector stored as .mat file (MATLAB), and the 'col' parameter in 'params_est_v5.m' denotes the batch size. I set it to 400x500 because constituent images in my training dataset were all resized to 400x500.
[mex_C++.zip](https://github.com/v1t0ry/ICAP/files/6138863/mex_C%2B%2B.zip)
